openstudiocore/src/openstudio_app/OpenStudioApp.cpp

```diff
@@ -113,748 +115,807 @@ using namespace openstudio::model;
+				
+				if (restoreTabs){
+					startTabIndex = m_osDocument->verticalTabIndex();
+					startSubTabIndex = m_osDocument->subTabIndex();
+				}
+
+				if( !closeDocument() ) { 
+					this->setQuitOnLastWindowClosed(wasQuitOnLastWindowClosed);
+					return false;
+				}
+				processEvents();
+			}
+
+			waitDialog()->setVisible(true);
+			processEvents();
+
+			m_osDocument = std::shared_ptr<OSDocument>( new OSDocument(componentLibrary(),
+																																	 hvacComponentLibrary(), 
+																																	 resourcesPath(), 
+																																	 model, 
+																																	 fileName, 
+																																	 false, 
+																																	 startTabIndex, 
+																																	 startSubTabIndex) );
+
+			connect(m_osDocument.get(), &OSDocument::closeClicked, this, &OpenStudioApp::onCloseClicked);
+			connect(m_osDocument.get(), &OSDocument::exitClicked, this, &OpenStudioApp::quit);
+			connect(m_osDocument.get(), &OSDocument::importClicked, this, &OpenStudioApp::importIdf);
+			connect(m_osDocument.get(), &OSDocument::importgbXMLClicked, this, &OpenStudioApp::importgbXML);
+			connect(m_osDocument.get(), &OSDocument::importSDDClicked, this, &OpenStudioApp::importSDD);
+			connect(m_osDocument.get(), &OSDocument::importIFCClicked, this, &OpenStudioApp::importIFC);
+			connect(m_osDocument.get(), &OSDocument::loadFileClicked, this, &OpenStudioApp::open);
+			connect(m_osDocument.get(), &OSDocument::osmDropped, this, &OpenStudioApp::openFromDrag);
+			connect(m_osDocument.get(), &OSDocument::loadLibraryClicked, this, &OpenStudioApp::loadLibrary);
+			connect(m_osDocument.get(), &OSDocument::newClicked, this, &OpenStudioApp::newModel);
+			connect(m_osDocument.get(), &OSDocument::helpClicked, this, &OpenStudioApp::showHelp);
+			connect(m_osDocument.get(), &OSDocument::aboutClicked, this, &OpenStudioApp::showAbout);
+			waitDialog()->setVisible(false);
+
+			m_startupView->hide();
+
+			versionUpdateMessageBox(versionTranslator, true, fileName, openstudio::toPath(m_osDocument->modelTempDir()));
+
+			this->setQuitOnLastWindowClosed(wasQuitOnLastWindowClosed);
+
+			return true;
+		}else{
+			LOG_FREE(Warn, "OpenStudio", "Could not open file at " << toString(fileName));
+
+			versionUpdateMessageBox(versionTranslator, false, fileName, openstudio::path());
+		}
+	}
+	return false;
 }
 
 void OpenStudioApp::buildCompLibraries()
 {
-  osversion::VersionTranslator versionTranslator;
-  versionTranslator.setAllowNewerVersions(false);
-
-  path p = resourcesPath() / toPath("MinimalTemplate.osm");
-  OS_ASSERT(exists(p));
-  boost::optional<Model> temp = versionTranslator.loadModel(p);
-  OS_ASSERT(temp);
-  m_compLibrary = temp.get();
-
-  p = resourcesPath() / toPath("hvaclibrary/hvac_library.osm");
-  OS_ASSERT(exists(p));
-  temp = versionTranslator.loadModel(p);
-  OS_ASSERT(temp);
-  m_hvacCompLibrary = temp.get();
+	osversion::VersionTranslator versionTranslator;
```

Comment:
@zhouchong90 can you change your indentation to 2 spaces according to our style guide